test/e2e_node/util.go

```diff
@@ -335,6 +336,11 @@ func getCRIClient() (internalapi.RuntimeService, internalapi.ImageManagerService
 	if err != nil {
 		return nil, nil, err
 	}
+	if framework.TestContext.CriProxyEnabled {
+		runtimeServiceProxy := criproxy.NewRuntimeServiceProxy(r)
+		imageServiceProxy := criproxy.NewImageServiceProxy(i)
+		return runtimeServiceProxy, imageServiceProxy, nil
```

Comment:
will this override the kubelet config's configured CRI endpoints?

Reply:
No, NewRuntimeServiceProxy and NewImageServiceProxy simply create wrappers around the real CRI.

like
```
	runtimeService = newInstrumentedRuntimeService(runtimeService)
	imageService = newInstrumentedImageManagerService(imageService)
```
in NewKubeGenericRuntimeManager

Reply:
but how we instruct kubelet process that we start to use these wrappers? Kubelet will be started in a separate process for the tests

Reply:
> but how we instruct kubelet process that we start to use these wrappers? Kubelet will be started in a separate process for the tests

I previously had a misunderstanding. I will modify my implementation: roughly, I will start the criproxy service before starting the kubelet, and then modify the original runtime endpoint so that calls to the CRI interface can be intercepted by criproxy.

Reply:
@SergeyKanzhelev Could you help confirm a few things? thx

1. Create a global `criproxy` object that mainly serves to start the `criproxy service` and inject faults.
2. Each test case injects delays or errors through the global `criproxy` object. Would this cause any impact between different test cases?"

Reply:
> Would this cause any impact between different test cases?

Yes it will. We will likely run these tests as "Serial" so they are not affecting each other. Later we may decide to mark some of them as non-Serial if the error injection they use is only injects error for a specific argument that is unique for the specific test

Reply:
> We will likely run these tests as "Serial" so they are not affecting each other. Later we may decide to mark some of them as non-Serial if the error injection they use is only injects error for a specific argument that is unique for the specific test

@SergeyKanzhelev Reimplemented the code following the above approach, please help check if there are any other issues. thanks